ownCloud/Settings/Passcode/PasscodeManager.swift

```diff
@@ -0,0 +1,375 @@
+//
+//  PasscodeManager.swift
+//  ownCloud
+//
+//  Created by Javier Gonzalez on 06/05/2018.
+//  Copyright © 2018 ownCloud GmbH. All rights reserved.
+//
+
+/*
+ * Copyright (C) 2018, ownCloud GmbH.
+ *
+ * This code is covered by the GNU Public License Version 3.
+ *
+ * For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+ * You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+ *
+ */
+
+import UIKit
+import ownCloudSDK
+
+class PasscodeManager: NSObject {
+
+    // MARK: - Interface view mode
+    enum PasscodeInterfaceMode {
+        case addPasscodeFirstStep
+        case addPasscodeSecondStep
+        case unlockPasscode
+        case unlockPasscodeError
+        case deletePasscode
+        case deletePasscodeError
+        case addPasscodeFirstStepAfterErrorOnSecond
+    }
+
+    // MARK: Global vars
+
+    // Common
+    private var window: LockWindow?
+    private let passcodeLength = 4
+    private let passcodeKeychainAccount = "passcode-keychain-account"
+    private let passcodeKeychainPath = "passcode-keychain-path"
+    private var passcodeMode: PasscodeInterfaceMode?
+    private var passcodeViewController: PasscodeViewController?
+
+    // Add/Delete
+    private var passcodeFromFirstStep: String?
+    private var completionHandler: CompletionHandler?
+
+    // Unlock
+    private var datePressedHomeButton: Date?
```

Comment:
Some other things could lead us to present the passcode like if you receive a call or the device locks itself because of inactivity, so maybe worth to change the name on this.

Reply:
I seted the name: `applicationWillResignActive` at the end it is what really happens